### ISSUE TITLE
[a11y] Text area: character count should be announced/read

### ISSUE THREAD

**Issue by aagonzales** [carbon-design-system/carbon#12071]
After reviewing documentation updates with @mbgower, some accessibility concerns were raised about features inside of text area. 

- Resize handle
- Character counter

[Stub]
We need to evaluate how these features work from an accessibility standpoint. I'm unsure of the exact next steps but wanted to open this issue so we don't forget about them.

**Comment by mbgower** [comment]
Native text areas do not include a keyboard method of expanding the text area. So Carbon's implementation is no _worse_ than the normal. But I don't think it should be _too_ difficult to add an additional tab stop on the resize icon, which gave the ability for the user to activate the resize functionality (user arrow keys). I'd like to hear from @tay1orjones or other FEDs about the feasibility of this. 

Possible model:
1. Press tab to exit Text area. Focus lands on Resize button.
2. User presses Enter/Spacebar to activate. When activated, maybe the cursor shape or button changes to a move  
![image](https://user-images.githubusercontent.com/14298245/188958221-565b1f07-5d8c-43d6-b8d7-3b0b8c346ce4.png)
 or resize icon.
![image](https://user-images.githubusercontent.com/14298245/188958283-ff3a0269-de95-40a8-8087-3f3ccaa80127.png)
3. User uses arrow keys to make wider or taller (or shorter/thinner). Presses Enter to keep, or Esc to quit without changing.

**Comment by mbgower** [comment]
The character count is tougher to achieve elegantly. I don't think anyone would want to have the character count announced all the time as they are typing. On the other hand, many users would likely want to be able to hear what the count is and maybe once down to a certain number of remaining characters (say 20) would then announce the character count.
 yer
I did a [quick search](https://www.google.com/search?q=accessible+character+counter&rlz=1C5GCEM_en___US996&oq=accessible+character+counter&aqs=chrome..69i57j0i390.6144j0j7&sourceid=chrome&ie=UTF-8) and several implementations exist. Haven't reviewed yet.

**Comment by jnm2377** [comment]
@mbgower the resizing idea is interesting but I don't see that implementation being feasible with the component as is. We're just using the native html `textarea` element under the hood which automatically handles the resizing/trigger. The only options we have are CSS styles, deciding whether to allow resizing vertically, horizontally, both or not at all. Since the handle isn't an actual element in the DOM, implementing a focusable resizing handle would likely require refactoring the textarea to be a custom component with custom resizing logic. Alternately, I think a much easier approach would be adding keyboard shortcuts to resize the textarea. What are your thoughts?

As far as the character count, I like your idea of announcing certain characters instead of at every change. I agree, that would probably be too much. Maybe announcing the changes at certain breakpoints like one-time 100% count (so users know how many characters they have), one-time 50% count and all of the last 20 or something like that?

**Comment by mbgower** [comment]
Okay, I will see if I can get some attention on the html textarea at the html/user agent level. I agree that if Carbon is just using textarea under the covers, this is not something we need to address.

For the character count, I like that idea of an initial note (May need to use aria-describedby; not sure if the polite live region will be announced immediately?) and then a one-time polite mention at 50%, with I'd say the last 10 characters going to an assertive setting for the live region. There are a couple of examples that use the switch from polite to assertive. This might be something we could test with some IBM screen reader users.

**Comment by dakahn** [comment]
a character count update seems like a role status to me (re: polite alert). With a 'no characters remaining' potentially being an assertive alert since there's potential user confusion when they can no longer type.

**Comment by tay1orjones** [comment]
We'll also want to add this to TextInput, https://github.com/carbon-design-system/carbon/issues/12429
